scala-effect-api/src/main/scala/com/avast/metrics/scalaeffectapi/impl/TimerPairImpl.scala

```diff
@@ -0,0 +1,26 @@
+package com.avast.metrics.scalaeffectapi.impl
+
+import cats.effect.Sync
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import java.time.Duration
+import com.avast.metrics.scalaeffectapi.{Timer, TimerPair}
+
+private class TimerPairImpl[F[_]: Sync](success: Timer[F], failure: Timer[F]) extends TimerPair[F] {
+  override def start: F[TimerPairContext] = {
+    for {
+      succCtx <- success.start
+      failCtx <- failure.start
+    } yield new TimerPairContext {
+      override def stop: F[Duration] = success.stop(succCtx)
+
+      override def stopFailure: F[Duration] = failure.stop(failCtx)
```

Comment:
i think it's simpler to introduce a case class for this but up to you

Reply:
yes, you can introduce a case class that contains the contexts and then call the stop method with the corresponding context but this looks better to me